### PR TITLE
refactor(web): move `model/types` to `web/types` 🏗️

### PR DESCRIPTION
**Pull request by ermshiperete**
Part-of: #12133

@keymanapp-test-bot skip

### CODE REVIEW

File: common/web/types/src/main.ts [hunk 1]
```diff
@@ -29,3 +29,5 @@ export * as KMXPlus from './kmx/kmx-plus/kmx-plus.js';
 export { UnicodeSetParser, UnicodeSet } from './ldml-keyboard/unicodeset-parser-api.js';
 export { VariableParser, MarkerParser } from './ldml-keyboard/pattern-parser.js';
 export { ElementString } from './kmx/kmx-plus/element-string.js';
+
+export * from './lexical-model-types.js';
```
Comment (mcdurdin) [inline common/web/types/src/main.ts:33]: Can we enumerate all the exports here? It's busy work but it helps us be explicit about what is a supported API surface. You'll note that we do not have other top-level global exports; they are all `* as xxxx`.

**Review by mcdurdin — APPROVED** [review]
LGTM. This needs to be documented as a breaking change for 18.0 -- API surface has significant refactor (albeit just moving things around, nothing new)

**Comment by ermshiperete** [comment]
@mcdurdin I added the change to the release notes section in the Google Doc. Does the breaking change has to be added elsewhere?

**Comment by mcdurdin** [comment]
> @mcdurdin I added the change to the release notes section in the Google Doc. Does the breaking change has to be added elsewhere?

That is all we need -- it'll make its way from there into the 18.0 release notes eventually

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 18.0.108-alpha](https://keyman.com/downloads/releases/alpha/18.0.108)

**Pull request merged**